src/app/messagelist/nativemessagelist.component.html

```diff
@@ -0,0 +1,53 @@
+<div id="nativemessagelist">
+    <div *ngIf="rowsDisplayed < rowCount">
+        Showing messages {{ offset + 1 }}..{{ upto }} out of {{ rowCount }}
+    </div>
+    <button mat-stroked-button class="pagingButton" *ngIf="offset > 0" (click)="scrollUp()">
+        Show previous {{ rowsDisplayed }} messages
+    </button>
+    <button mat-stroked-button
+        *ngFor="let row of shownRows; index as i"
+        [ngClass]="{ 'messageButton': true, 'selectedFolder': rows.openedRowIndex == row.rowid }"
+        (click)="openMessage(row.rowid)"
+    >
+        <div class="firstRow">
```

Comment:
Looks like this hard codes the set of columns displayed - which won't quite DWIM when showing search results, or the "Sent" folder.. could it use the canvastablecolumns stuff somehow?

Reply:
It does, though it cheats a little by hardcoding the header column to be either From or Sent. Not very future-proof, but that's the best we've got right now I think.